### ISSUE TITLE
Cannot parse "-"

### ISSUE THREAD

**Issue by emmatebibyte** [DoumanAsh/arg.rs#2]
When passing "-" to a command to give it stdin, the command returns:
`'target/x86_64-unknown-linux-gnu…' terminated by signal SIGILL (Illegal instruction)`

**Comment by DoumanAsh** [comment]
huh, interesting

Can you show your arguments struct?

**Comment by DoumanAsh** [comment]
It would be ideally great if you already have example I could use to reproduce issue

**Comment by DoumanAsh** [comment]
Btw, I don't think `arg` code can segfault, if you pass just `-` it should fail on invalid argument
There is no special treatment for `--` to propagate arguments somewhere

Unless `exit` function can segfault I guess
https://github.com/DoumanAsh/arg.rs/blob/master/src/lib.rs#L182

**Comment by emmatebibyte** [comment]
Here's my code:
```rs
use std::env;
use std::fs::{File, read_to_string};
use std::io;
use std::io::{ BufRead, Write };
use std::os::unix::io::FromRawFd;

use arg::Args;

use sysexits::ExitCode;

#[derive(Args, Debug)]
struct Arguments {
  #[arg(short = "u")]
  u: bool,

  paths: Vec<String>,
}

fn main() -> ExitCode {
  let mut arguments = env::args().collect::<Vec<String>>();
  let argv0 = arguments.remove(0);

  let mut args = Arguments::from_args(arguments
    .iter()
    .map(String::as_str)
    .collect::<Vec<&str>>()
  ).unwrap();

  let mut val = String::new();
  let mut output = String::new();

  if args.paths.is_empty() { args.paths.push("-".to_string()); }

  for path in args.paths {
    if path == "-" {
      match io::stdin().lock().read_line(&mut output) {
        Ok(_) => { val.push_str(&output); },
        Err(_) => {
          eprintln!("Usage: {} [options...] [files...]", argv0);
          return ExitCode::Usage;
        },
      };
    } else {
      match read_to_string(&path) {
        Ok(output) => { val.push_str(&output); },
        Err(_) => {
          eprintln!("{}: {}: No such file or directory.", argv0, &path);
          return ExitCode::NoInput;
        },
      };
    }
  }
  if args.u {
    let mut stdout = unsafe { File::from_raw_fd(1) };
    match stdout.write_all(&val.as_bytes()) {
      Ok(_) => {},
      Err(_) => {
        eprintln!("{}: Cannot write to stdout", argv0);
        return ExitCode::OsErr;
      },
    };
  } else { print!("{}", &val); }

  match io::stdout().flush() {
    Ok(_) => return ExitCode::Ok,
    Err(_) => {
      eprintln!("{}: Cannot write to stdout", argv0);
      return ExitCode::OsErr;
    },
  };
}
```

**Comment by emmatebibyte** [comment]
For reference I'm using nightly with a bunch of custom build stuff:
```toml

[profile.release]
strip = true  # strip symbols from the binary
opt-level = "z" # optimize for size
lto = true # link time optimization
codegen-units = 1 # decrease parallelization
panic = "abort"
```
```sh
#!/bin/sh

cargo build --release \
	-Z build-std=std,panic_abort \
	-Z  build-std-features=panic_immediate_abort \
	--target x86_64-unknown-linux-gnu
```

**Comment by DoumanAsh** [comment]
I see so segfault is because it panics on `unwrap` and you set panic as abort

In any case since there is no segfault
I just want to clarfiy how do you want to handle `-`
Do you want just store it into `paths`?
I'm asking because I'm not sure how better design handling of `-` and `--` alone

Btw normally CLI parsers use `--` to forward arguments to another program

**Comment by emmatebibyte** [comment]
> Do you want just store it into `paths`?
Yep :) because I'm writing POSIX shell utilities using your crate:
from `man 1p cat`:
```
OPERANDS
       The following operand shall be supported:

       file      A  pathname of an input file. If no file operands are specified, the standard input shall be used. If a
                 file is '-', the cat utility shall read from the standard input at that point in the sequence. The  cat
                 utility  shall  not close and reopen standard input when it is referenced in this way, but shall accept
                 multiple occurrences of '-' as a file operand.

STDIN
       The standard input shall be used only if no file operands are specified, or if a file operand is  '-'.   See  the
       INPUT FILES section.

INPUT FILES
       The input files can be any file type.
```

**Comment by DoumanAsh** [comment]
ah I see, I'll look into it after work, I'll let you know once I fix derive code

**Comment by emmatebibyte** [comment]
Thank you so much! :)

**Comment by DoumanAsh** [comment]
I published new version of `arg-derive` 0.3.5
update your dep please
